From 380d68bb2eeafd4b7b562a6645f9a29a8da1c719 Mon Sep 17 00:00:00 2001
From: Cody Peter Mello <cody.mello@joyent.com>
Date: Fri, 24 Aug 2018 22:24:22 +0000
Subject: [PATCH] MORAY-490 Electric Moray should pass through options for
 bucket operations Reviewed by: Jan Wyszynski <jan.wyszynski@joyent.com>
 Reviewed by: Jared Morrow <jm@joyent.com> Approved by: Jared Morrow
 <jm@joyent.com>

---
 lib/server.js | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/lib/server.js b/lib/server.js
index c019a11..fdaa7fe 100644
--- a/lib/server.js
+++ b/lib/server.js
@@ -481,9 +481,8 @@ function getBucket(options) {
             }
 
             var pnode = hNode.pnode;
-            options.clients.map[pnode].getBucket(bucket,
-                                                 function (gErr, rbucket)
-            {
+            options.clients.map[pnode].getBucket(bucket, opts,
+                function (gErr, rbucket) {
                 log.debug({
                     err: gErr,
                     bucket: rbucket
@@ -573,7 +572,7 @@ function updateBucket(options) {
         }
 
         options.clients.array.forEach(function (client) {
-            client.updateBucket(name, cfg, function (err2) {
+            client.updateBucket(name, cfg, opts, function (err2) {
                 log.debug({
                     err: err2
                 }, 'updateBucket: returned');
@@ -647,7 +646,7 @@ function delBucket(options) {
         }
 
         options.clients.array.forEach(function (client) {
-            client.delBucket(name, function (err2) {
+            client.delBucket(name, opts, function (err2) {
                 log.debug({
                     err: err2
                 }, 'delBucket: returned');
-- 
2.21.0

